test/libsolutil/StringUtils.cpp

```diff
@@ -210,6 +210,38 @@ BOOST_AUTO_TEST_CASE(test_format_number_readable_signed)
 	);
 }
 
+BOOST_AUTO_TEST_CASE(test_encodeURI)
+{
+	BOOST_CHECK_EQUAL(util::encodeURI(""), "");
+	BOOST_CHECK_EQUAL(util::encodeURI(" "), "%20");
+	BOOST_CHECK_EQUAL(util::encodeURI("%"), "%25");
+	BOOST_CHECK_EQUAL(util::encodeURI("Hello World."), "Hello%20World.");
+	BOOST_CHECK_EQUAL(util::encodeURI("\x01\x02\x7F"), "%01%02%7F");
+	BOOST_CHECK_EQUAL(util::encodeURI("C:\\readme.md"), "C:%5Creadme.md");
+	BOOST_CHECK_EQUAL(util::encodeURI("C:/readme.md"), "C:/readme.md");
+	BOOST_CHECK_EQUAL(util::encodeURI("/var/log"), "/var/log");
```

Comment:
Could you add a few more cases?
- A Windows path with a `file://` prefix.
- `😃`
- `%%`.